backend/routes/matchRouter.js

```diff
@@ -0,0 +1,50 @@
+// handle saved games
+const express = require("express");
+const router = express.Router();
+
+const matchModel = require("../models/matchModel.js");
+const userModel = require("../models/userModel.js");
+
+// routes: upload match, get match history
+
+router.post("/uploadMatch", (req, res) => {
```

Comment:
This can be used for daily challenges; room games, for what I have in mind, decide whether to terminate a game in the backend (since it will not be synchronous). We can always introduce the concept of a "room owner" who has the right to close the room, though.

Reply:
Since this is meant for each "match" (assuming a room can have multiple matches), this would be updated at the end of each match/round. For the daily challenge this is pretty straight forward.